src/box/tuple.h

```diff
@@ -170,6 +170,26 @@ box_tuple_format(box_tuple_t *tuple);
 const char *
 box_tuple_field(box_tuple_t *tuple, uint32_t fieldno);
 
+/**
+ * Return the raw tuple field pointed by a full JSON path in
+ * MsgPack format.
+ *
+ * The JSON path includes the outmost field. For example, "c" in
+ * ["a", ["b", "c"], "d"] can be accessed using '[2][2]' path. The
+ * array element accessors are one-based (as in Lua).
+ *
+ * The first JSON path token may be a field name if the tuple
+ * has associated format with named fields.
+ *
+ * The buffer is valid until next call to box_tuple_* functions.
+ *
+ * Return NULL if the field does not exist or if the JSON path is
+ * malformed or invalid. Multikey JSON path token [*] is treated
+ * as invalid in this context.
+ */
+API_EXPORT const char *
+box_tuple_field_by_full_path(box_tuple_t *tuple, const char *path,
+			     const char *path_end);
```

Comment:
Why not pass path length instead, like `box_space_id_by_name` does? This would allow one to pass string literals to the function without introducing variables:

```C
const char *field = box_tuple_field_by_path(tuple, "foo.bar", 7);
```

Reply:
I was under impression that our module API usually accepts `const char *foo` and `const char *foo_end` like in `box_tuple_new()`. However, it seems, it is about msgpack data and has no relation to names and so on.

I has no strict opinion what is the best choice, so I'll follow your suggestion.

Reply:
Changed from `const char *path_end` to `uint32_t path_len`.

Maybe `size_t` is more appropriate here, but we already have `uint32_t` in `box_{space,index}_id_by_name()` and there is no difference in practice.